### ISSUE TITLE
cmd field never checked in the server

### ISSUE THREAD

**Issue by bitlogik** [Scille/parsec-cloud#1626]
The server doesn't care about the content in cmd field, and the string can be whatever. Also this field can be undefined with the cmd key not provided. It is like this "cmd" field is totally useless for the server.

Example : sending cmd:"ping" instead of cmd:"handshake"
```
--> {"cmd":"ping", "client_api_version":[2,1], "handshake":"answer", "answer":[...], "type":"AUTHENTICATED", "organization_id":"myOrg", "rvk":[...], "device_id":"..."}
<-- {"handshake":"result", result: "ok"}
```
Same without the cmd key.

I feel like "how on earth could we miss that?". What is the purpose of this cmd field? If this is useless, better to remove it. But I would suggest to add its read and check of this field, unless it was part of a abandon plan.

**Comment by touilleMan** [comment]
When opening a connection with the backend, the client has to first do a handshake phase

1 - server sending a `handshake: challenge` message
2 - client answering a `handshake: answer` message, this return the kind of handshake the client wants to do (anonymous or authenticated)
3 - server finishing with a `handshake: result` message to confirm everything is ok

Only once the handshake has been done the client can send commands to the server.

In your example you got tricked by the fact the server ignore unknown fields when doing message deserialization (this is done to allow backward/forward compatibility).

**Comment by bitlogik** [comment]
Yep, we double checked, and indeed, there is no *cmd* key in [the handshake command](https://github.com/Scille/parsec-cloud/blob/5db9aec1935797cb10d7919a0a390c98b5f34889/parsec/api/protocol/handshake.py#L409). Turns out that at some point during the development of our new web-based client app, we added "cmd":"handshake" in the handshake query. The server just ignores additional keys, and mutating the content doesn't change anything. Thanks for your help, we'll remove in our code this key field, as this is unknown by the protocol at this step.
